geomesa-feature/src/main/scala/org/locationtech/geomesa/feature/EncodingOption.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2015 Commonwealth Computer Research, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.locationtech.geomesa.feature
+
+/**
+ * Options to be applied when encoding.  The same options must be specified when decoding.
+ */
+object EncodingOption extends Enumeration {
+  type EncodingOption = Value
+
+  /**
+   * If this [[EncodingOption]] is specified then all user data of the simple feature will be
+   * serialized and deserialized.
+   */
+  val WITH_USER_DATA = Value("withUserData")
```

Comment:
I would leave out the string arg to Value - that way the value will match the enum name, which is more user-friendly. Also, the convention in scala is to use camel case for constants/enums - e.g. WithUserData (not sure if we have followed that everywhere though...)